internal/services/newrelic/new_relic_monitor_resource.go

```diff
@@ -265,6 +268,11 @@ func (r NewRelicMonitorResource) Create() sdk.ResourceFunc {
 				},
 			}
 
+			identityValue := expandSystemAssigned(metadata.ResourceData.Get("identity").([]interface{}))
+			if identityValue.Type != identity.TypeNone {
+				properties.Identity = identityValue
+			}
```

Comment:
This isn't how the `Type` field in `Identity` is usually handled which would imply a bug, this should also be raised as an issue on the Rest API specs repo.

Reply:
created a issue to track this problem:  https://github.com/Azure/azure-rest-api-specs/issues/29257

Reply:
In the future: can we also update the API Definition to expose the `Identity` field correctly too? We've spent a _considerable_ amount of time standardising the behaviours (and methods) for the `Identity` types, so we should lean on them where possible - as such I've opened https://github.com/hashicorp/pandora/pull/4171 which'll allow this.